vm/neovm/types/neovm_value.go

```diff
@@ -241,21 +259,26 @@ func (self *VmValue) convertNeoVmValueHexString(count *int) (interface{}, error)
 		var sstr []interface{}
 		for i := 0; i < len(self.array.Data); i++ {
 			*count++
-			t, err := self.array.Data[i].convertNeoVmValueHexString(count)
+			t, err := self.array.Data[i].convertNeoVmValueHexString(count, length)
 			if err != nil {
 				return nil, err
 			}
 			sstr = append(sstr, t)
 		}
 		return sstr, nil
 	case interopType:
-		return common.ToHexString(self.interop.Data.ToArray()), nil
+		bs := self.interop.Data.ToArray()
+		*length += len(bs)
+		return common.ToHexString(bs), nil
 	default:
 		panic("unreacheable!")
 	}
 }
 
-func (self *VmValue) Deserialize(source *common.ZeroCopySource) error {
+func (self *VmValue) Deserialize(source *common.ZeroCopySource, depth *int) error {
```

Comment:
public api should not be changed for easier usage. wrap the depth check logic in private method.